src/Microsoft.Extensions.DependencyInjection/ServiceProvider.cs

```diff
@@ -15,7 +14,7 @@ namespace Microsoft.Extensions.DependencyInjection
     /// <summary>
     /// The default IServiceProvider.
     /// </summary>
-    internal class ServiceProvider : IServiceProvider, IDisposable
+    public sealed class ServiceProvider : IServiceProvider, IDisposable
```

Comment:
Can't add extra data to the type and then use an `as` cast to surface it with a wrap...